src/main/asciidoc/entity-callbacks.adoc

```diff
@@ -0,0 +1,163 @@
+[entity-callbacks]
+= Entity Callbacks
+
+The Spring Data infrastructure provides hooks for modifying an entity before and/or after certain methods are invoked.
+Those so called ``EntityCallback``s provide a convenient way to check and potentially modify an entity in a callback fashioned style. +
+An `EntityCallback` looks pretty much like a specialized `ApplicationListener`, with which you might be already familiar.
+Some Spring Data modules publish store specific events, like a `BeforeSaveEvent` that allow modifying the given entity which in some cases, eg. immutable types, can cause trouble.
+Another aspects is that, event publishing relies on `ApplicationEventMulticaster` that can be configured with an asynchronous `TaskExecutor` leading to unpredictable outcome as event processing can be forked onto a `Thread`.
```

Comment:
I would rewrite this one to be: "Also, event publishing relies on `ApplicationEventMulticaster`. Beware of configuring that with an asynchronous `TaskExecutor`. Doing so can lead to unpredictable outcomes, as event processing can be forked onto a `Thread`."